extensions-contrib/opentelemetry-emitter/src/main/java/org/apache/druid/emitter/opentelemetry/DruidContextTextMapGetter.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.emitter.opentelemetry;
+
+import io.opentelemetry.context.propagation.TextMapGetter;
+
+import java.util.Map;
+
+public class DruidContextTextMapGetter implements TextMapGetter<Map<String, String>>
+{
+  @Override
+  public String get(Map<String, String> carrier, String key)
+  {
+    if (carrier.containsKey(key)) {
+      return carrier.get(key);
+    }
+    return null;
+  }
+
+  @Override
+  public Iterable<String> keys(Map<String, String> carrier)
+  {
+    return carrier.keySet();
```

Comment:
are we sure we want to use all the keys in the druid context? It might make sense to exclude context keys that are only used internally for query processing purposes.

Reply:
I think we can add `opentelemetryContext` field into druid `context`  where we would store everything about opentelemetry

Reply:
As far as I can see, druid uses only needed keys(like `traceparent` and `tracestate`) to propagate trace context and doesn't use other keys

Reply:
Added in README